---
title: "Project Update"
author: "Brad Blaser"
date: "8/23/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Fig 1

Cell type identification by expression of specific markers.

### Fig 1A

UMAP representation of all cells (N = 103,415) in the final dataset after filtering for doublets and low-quality cells and batch correction according to patient identity.  Each patient is represented by a different color.

### Fig 1B

UMAP representation of all cells in the final dataset after clustering using the partitioning method.  Cell clusters were identified by inspection of top markers according to their Jensen-Shannon divergence.

### Fig 1C

Selected top cluster markers.

### Fig 1D

Expression of AML blast markers and canonical lymphocyte markers by scRNA-seq cell cluster.

```{r, dev='png', fig.width=7.5, fig.height=8, dpi=300}
fig1
```

## Supplemental Fig 1

### Supplemental Fig 1A

UMAP representation of all cells prior to batch correction.

### Supplemental Fig 1B

UMAP representation of all cells prior to batch correction.  Colors represent cluster assignments performed after batch correction according to patient identity.

### Supplemental Fig 1C

Representation within cell clusters according to patient identity . Color scale within each cell represents the log10-transformed percent of each cluster represented by that sample.  Row totals are each equal to 100 percent. 

```{r, dev='png', fig.width=5.5, fig.height=10, dpi=300}
sup_fig1
```


## Fig2

Differential gene expression.

### Fig 2A

Expression of HGF by cell clusters, stratified by treatment day and response to ficlatuzumab.

### Fig 2B

Expression of CSF1R by cell clusters, stratified by treatment day and response to ficlatuzumab.

### Fig 2C

Day 0 myeloid cell clusters (Early blasts, HLA-DR+, and Late Blasts) were identified and differential gene expression was performed using pseudobulk analysis (DESeq2).  HGF, CSF1R and selected genes are indicated.  Genes with FDR-adjusted p values less than 0.05 and log2fc greater than 1.5 are indicated in red.

### Fig 2D

Day 42-44 myeloid cell clusters were identified and pseudobulk analysis was performed as in 2C.

```{r, dev='png', fig.width=7.5, fig.height=6, dpi=300}
fig2
```

## Supplemental Fig 2

UMAP representation of expression of selected genes.

### Supplemental Fig 2A

HGF expression stratified by treatment day and response to ficlatuzumab.  Colors represent log10-transformed, size-factor normalized expresison prior to performing batch correction. 

### Supplemental Fig 2B

CSF1R expression stratified by treatment day and response to ficlatuzumab.  Colors represent log10-transformed, size-factor normalized expresison prior to performing batch correction. 

```{r, dev='png', fig.width=7.5, fig.height=6, dpi=300}
sup_fig2
```

## Fig 3

Gene module analysis identifies viral response and protein translation genes as potential biomarkers of resistance to ficlatuzumab.

### Fig 3A

Gene modules were identified by unsupervised analysis.  Expression of module genes according to treatment day and response to ficlatuzumab is shown.

### Fig 3B

Aggregated counts for module 7 genes in responders and non-responders on each treatment day are shown.

### Fig 3C

Gene Ontology terms enriched in module 7 genes.

### Fig 3D

Aggregated counts for module 18 genes in responders and non-responders on each treatment day are shown.

### Fig 3E

Gene Ontology terms enriched in module 18 genes.

```{r, dev='png', fig.width=7.5, fig.height=10, dpi=300}
fig3
```


## Cytof-scRNAseq correlation

Compared to the prior version, I removed some markers that didn't correlate well.  Annotation color bars on the top show the assay (cytof vs scrnaseq) and the cell type used to generate the expression values.

Greg provided mean cytof expression values for these celltypes for each sample he ran (it was only a subset of the available samples). I added Pan-hematopoietic after the fact because CD45 was in there.  I created analogous clusters to Gregs from the scRNA seq expression data.  Then I identified corresponding transcripts for the cytof markers e.g. 168Er_CD8a<=>CD8A and aggregated the RNA expression of each marker within the previously defined clusters.  I arranged the cytof and scrnaseq markers in side by side and did a pairwise correlation across all possible combinations of markers.  Colors indicate Pearson correlation coefficient.

It looks like scrnaseq and cytof are lining up pretty well now and forming distinct clusters according to what we expected.

```{r, dev='png', fig.width=7.5, fig.height=6, dpi=300}
cor_heatmap
```

